app/src/main/java/com/xinto/mauth/ui/component/pinboard/PinBoard.kt

```diff
@@ -29,54 +31,140 @@ fun PinBoard(
     modifier: Modifier = Modifier,
     state: PinBoardState = rememberPinBoardState()
 ) {
-    FlowRow(
-        modifier = modifier,
-        verticalArrangement = Arrangement.spacedBy(8.dp, Alignment.CenterVertically),
-        horizontalArrangement = Arrangement.spacedBy(16.dp, Alignment.CenterHorizontally),
-        maxItemsInEachRow = 3
-    ) {
-        state.buttons.forEach { button ->
-            when (button) {
-                is PinBoardState.PinBoardButton.Number -> {
-                    PinButton(
-                        modifier = Modifier.weight(1f),
-                        onClick = { state.onNumberClick(button.number) }
-                    ) {
-                        Text(button.toString())
+    val orientation = LocalConfiguration.current.orientation
+    if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
+
+        FlowColumn(
+            modifier = modifier,
+            verticalArrangement = Arrangement.spacedBy(8.dp, Alignment.CenterVertically),
+            horizontalArrangement = Arrangement.spacedBy(16.dp, Alignment.CenterHorizontally),
+            maxItemsInEachColumn = 4
+        ) {
+            val rowCount = 4
+            val rowWidth = 3
+            val columnCount = 3
+            val columnButtons = buildList {
+                for (columnIndex in 0..<columnCount) {
+                    for (rowIndex in 0..<rowCount) {
+                        add(state.buttons[rowWidth * rowIndex + columnIndex])
                     }
                 }
-                is PinBoardState.PinBoardButton.Backspace,
-                is PinBoardState.PinBoardButton.Fingerprint,
-                is PinBoardState.PinBoardButton.Enter -> {
-                    PrimaryPinButton(
-                        modifier = Modifier.weight(1f),
-                        onClick = when (button) {
-                            is PinBoardState.PinBoardButton.Backspace -> state.onBackspaceClick
-                            is PinBoardState.PinBoardButton.Fingerprint -> state.onFingerprintClick
-                            is PinBoardState.PinBoardButton.Enter -> state.onEnterClick
-                            else -> throw NoSuchElementException()
-                        },
-                        onLongClick =
+            }
+            columnButtons.forEach { button ->
+                when (button) {
```

Comment:
We cannot extract this common code into a `Composable` since it uses modifiers like `weight` which is an extension function that changes based on `FlowRow` or `FlowColumn` scope.